### PR TITLE
Defibrilator and Defibrilator HUD bug fixes

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
The medical HUDs will report an icon when someone is defibbable. This occasionally will report inconsistent results, meaning its usefulness is diminished. It will sometimes report bodies as defibbable even though they have committed suicide, for example. This PR fixes this--if it is impossible for a carbon to be recovered, it will show as a skull. If it is somehow possible to be defibbed (such as simply having a lot of brute/burn damage), it will correctly show the defib icon.

This code also changes the return type of `/mob/living/carbon/proc/can_defib` to return either TRUE or a bit flag detailing why the carbon cannot be defibbed. This is then used to remove code that duplicates these checks to tell the user why the defibrilation failed.

## Why It's Good For The Game

Doctors should be able to trust the medical HUD.

## Changelog
:cl:
fix: Fixed the medical HUD sometimes reporting undefibbable bodies as defibbable and vice versa.
fix: Fixed plasmamen needing hearts in order to be defibbed.
fix: Fixed crew with HARS needing a brain to be defibbed.
/:cl:

### CODE REVIEW

**Comment by MrPerson** [comment]
Rather than use TRUE, I think you should #define DEFIB_POSSIBLE (1<<0) and use that.

**Comment by Jared-Fogle** [comment]
Sure.

File: code/modules/research/nanites/nanite_programs/healing.dm [hunk 1]
```diff
@@ -224,8 +224,8 @@
 		return FALSE
 	var/mob/living/carbon/C = host_mob
 	if(C.get_ghost())
-		return FALSE
-	return C.can_defib()
+		return DEFIB_POSSIBLE
```
Comment (Jared-Fogle) [inline code/modules/research/nanites/nanite_programs/healing.dm:227]: This looks like a mistake. I will fix this.

**Comment by MrMelbert** [comment]
Previously being husked triggered the "Tissue Damage" warning, now it looks like it triggers the "Heart Damage" warning. Is this change intended?

**Comment by Jared-Fogle** [comment]
@MrMelbert No, I will fix this as well.

**Comment by ATH1909** [comment]
can you also make it so that you don't have to punch people who've left the game AFTER they died in order to update their medHUD icon? it's currently a part of my healing routine to punch a corpse before trying to rez it, and a decent amount of the time, the defib icon immediately turns into a skull when I do so.

**Comment by Jared-Fogle** [comment]
Left the game as in closing out the BYOND window? I'll look into that and see if its already fixed by this, and fix it otherwise.

**Comment by Jared-Fogle** [comment]
Should be fixed?

**Comment by Timberpoes** [comment]
While you're working on defib code, can you look at certain weird edge cases:

HARS mutation makes the brain a non-essential organ. As a result, a player can be alive without a brain (or even an entire head) but then cannot be revived by defibs.

Plasmapeople don't have hearts and thus can't be revived without implanting an utterly useless heart their body doesn't use into them.

There's probably more weird edge-casery, but those are two that occur off the top of my head.

On the other hand, that shitcode allows things like implanting a bunch of random organs into skeletons to revive them. So I'm kinda torn.

**Comment by Jared-Fogle** [comment]
I can fix those two specifically, sure. I'm not aware of any other edge cases.

**Comment by ATH1909** [comment]
> I can fix those two specifically, sure. I'm not aware of any other edge cases.

Did you fix those by making them work, or by making them NOT work?

**Comment by Jared-Fogle** [comment]
@ATH1909 They did not defib with either the current system or my refactor. Now, they do.

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**